src/simple_cache_client.rs

```diff
@@ -31,11 +31,11 @@ use crate::utils;
 use crate::{grpc::header_interceptor::HeaderInterceptor, utils::connect_channel_lazily};
 use crate::{utils::user_agent, MomentoResult};
 
-pub trait IntoBytes {
+pub trait IntoBytes: Send {
```

Comment:
@kvcache I also meant to check in with you to make sure this is sane. I think it is, but wanted to doublecheck.

Reply:
Requiring Send on IntoBytes is reasonable - but you could avoid the constraint by eagerly transforming to bytes, before you fork a Send context (before you go async). I think it's okay to require Send for these; the 1 thing I checked to make sure there wouldn't be any obvious problems in the medium term future is https://docs.rs/bytes/1.5.0/bytes/struct.Bytes.html#impl-Send-for-Bytes (it's also implemented for BytesMut). I'm not concerned about this constraint.

Reply:
If we stick with the pattern where the trait has (effectively) an `async fn send` on Self, then the only place I can see where I could call `into_bytes` before we find ourselves inside of `async fn send` would be in `::new`. Meaning that the request structs themselves would all end up with `Vec<u8>` for all the things, rather than generic `IntoBytes` fields. I think that would probably be mostly okay but it might be annoying in debugging situations if someone was trying to log a request object that would have otherwise had `String` fields.

I think the UX win of supporting generics -> Strings in those structs is probably worth requiring `: Send` on IntoBytes but lmk if you agree.

Reply:
Debuggers tend to be amenable to printing conveniently-ascii byte sequences as ascii strings in my experience. Also, you don’t have a `std::fmt::Debug` bound on IntoBytes, so nobody is doing the log version of that story today or with the code in this pr anyway ;-) right? I do not really think that debugger pattern matters very much. Removing the bound and storing a plain vec might make some code simpler to read, which is maybe even better than a possible luxury for an attached debugger.